searchlib/src/vespa/searchlib/attribute/enum_store_dictionary.cpp

```diff
@@ -58,19 +50,27 @@ EnumStoreDictionary<BTreeDictionaryT, HashDictionaryT>::free_unused_values(const
             _enumStore.free_value_if_unused(iter.getKey(), unused);
         }
     } else {
-        this->_hash_dict.foreach_key([this, &unused](EntryRef ref) { _enumStore.free_value_if_unused(ref, unused); });
+        this->_hash_dict.foreach_key([this, &unused](EntryRef ref) {
+            _enumStore.free_value_if_unused(ref, unused);
+        });
     }
     remove_unused_values(unused, cmp);
 }
 
 template <typename BTreeDictionaryT, typename HashDictionaryT>
 void
 EnumStoreDictionary<BTreeDictionaryT, HashDictionaryT>::free_unused_values(const IndexSet& to_remove,
-                                                     const vespalib::datastore::EntryComparator& cmp)
+                                                                           const vespalib::datastore::EntryComparator& cmp)
 {
     IndexSet unused;
+
+    EntryRef prev;
     for (const auto& index : to_remove) {
-        _enumStore.free_value_if_unused(index, unused);
+        assert(prev <= index);
+        if (index != prev) {
```

Comment:
Is this check really needed when _to_remove_ is still an _IndexSet_?

Reply:
Deceived, by the naming. Multilevel typedefs can be deceiving. I will clean up.

Reply:
Now IndexSet should be fully replaced by IndexList.